darkcluster/src/main/java/com/linkedin/darkcluster/filter/DarkClusterResponseValidationMetricsReaderFilter.java

```diff
@@ -0,0 +1,56 @@
+package com.linkedin.darkcluster.filter;
+
+import com.linkedin.darkcluster.api.DarkClusterResponseValidationMetricsCollector;
+import com.linkedin.darkcluster.api.ResponseValidationMetricsHeader;
+import com.linkedin.darkcluster.api.DarkClusterConstants;
+import com.linkedin.r2.filter.NextFilter;
+import com.linkedin.r2.filter.message.rest.RestFilter;
+import com.linkedin.r2.message.RequestContext;
+import com.linkedin.r2.message.rest.RestRequest;
+import com.linkedin.r2.message.rest.RestResponse;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A filter that is to be enabled only in a dark cluster that is meant to handle response validation metrics
+ * sent from source
+ */
+public class DarkClusterResponseValidationMetricsReaderFilter implements RestFilter {
+  private final DarkClusterResponseValidationMetricsCollector _metricsCollector;
+  private final ExecutorService _executorService;
+
+  private static final Logger LOG = LoggerFactory.getLogger(DarkClusterResponseValidationMetricsReaderFilter.class);
+
+  public DarkClusterResponseValidationMetricsReaderFilter(
+      DarkClusterResponseValidationMetricsCollector metricsCollector,
+      ExecutorService executorService) {
+    _metricsCollector = metricsCollector;
```

Comment:
check for null in the constructor. If null,  have a no-op/default implementation, so you don't need to check for null later. or, if appropriate, add the Nonnull annotations.